hedera-node/src/main/java/com/hedera/services/legacy/stream/RecordStream.java

```diff
@@ -466,21 +472,28 @@ public static boolean isRecordSigFile(File file) {
 	}
 
 	/**
-	 * Read the previous file hash from file system
-	 * @param directory
-	 * @return
+	 * Read the previous file hash from the last .rcd_sig file in the given directory
+	 *
+	 * @param directory the directory where stores record stream v3 files
+	 * @return record stream v3 previous file hash byte array
 	 */
 
-	private byte[] readPrevFileHash(String directory) {
-		File dir = new File(directory);
-		File[] files = dir.listFiles();
-		Optional<File> lastSigFileOptional = Arrays.stream(files).filter(file -> isRecordSigFile(file))
-				.max(Comparator.comparing(File::getName));
-		if (lastSigFileOptional.isPresent()) {
-			File lastSigFile = lastSigFileOptional.get();
-			return getFileHashFromSigFile(lastSigFile);
+	public static byte[] readPrevFileHash(String directory) {
+		if (directory != null) {
```

Comment:
nit : Instead of multiple `if` conditions inside if , it will be better to return `return new byte[DigestType.SHA_384.digestLength()];` if `directory == null` initially.

Reply:
Thank you @Neeharika-Sompalli .
because if `directory == null`, or the directory doesn't contain any file, or there is not any .rcd_sig file in this directory, we will `return new byte[DigestType.SHA_384.digestLength()];`